sched/scheduler/scheduler.go

```diff
@@ -8,6 +10,10 @@ import (
 	"sync"
 )
 
+type Scheduler interface {
```

Comment:
Yea I viewed this as Jason described.  The scheduler should schedule work, where that work come froms shouldn't matter.  Also The deadletter will have to get handled somewhere.  The WorkGenerator can evolve eventually to do this.